### ISSUE TITLE
Docker-Compose Broker broken. Depends on Docker link Feature that is not there anymore.

### ISSUE THREAD

**Issue by CloudResearch** [eclipse/kapua#1464]
Running Broker using the official docker-compose.yaml under dev-tools, which indeed points to docker-hub images, will result in an failure in operation. ES_PORT_9200_TCP_ADDR, ES_PORT_9200_TCP_PORT are not set after starting container, because in Docker-Compose Version 3.1, depends_on set no environment variables as link did in former days (see,  https://docs.docker.com/compose/compose-file/#links) (same for DB_PORT/ADDR variable). The container itself can build, but at runtime it failed to connect to DB/ES after 2 minutes and crashes. Any suggestions how to fix this problem?

**Comment by lorthirk** [comment]
Hi @CloudResearch, 

Could you tell me the tags of the images downloaded by docker compose? a simple `docker images kapua/*` should be enough.

**Comment by CloudResearch** [comment]
kapua/kapua-console               latest              9fb14526e47d        2 months ago        589MB
kapua/kapua-api                   latest              9869387dbe5c        2 months ago        545MB
kapua/kapua-broker                latest              309f20d37ca9        2 months ago        558MB
kapua/kapua-sql                   latest              b643a13120b7        2 months ago        432MB
kapua/kura-simulator              latest              30a9e6904251        7 months ago        114MB

**Comment by lorthirk** [comment]
Thanks for the feedback. As suspected `latest` tag is still pointing to 0.3.2 version, that indeed relied on those env variables. Using tag `1.0.0-M1` or `1.0.0-M2` (that we're pushing to Docker Hub in these very minutes) shouldn't use those variables anymore. My suggestion is to wait a few minutes for `1.0.0-M2`, so you can run your containers on updated images; anyway `1.0.0-M1` should be fine as well.

**Comment by CloudResearch** [comment]
Thanky very much for the fast feedback, so everything should work as expected now?

**Comment by lorthirk** [comment]
Yes. The images still try to declare env variables using the "link-generated" ones, but then they're not used anymore, so you should have no issues in running them.

In the meanwhile we released 1.0.0-M2 in Docker Hub, so you can set such version in `dev-tools/src/main/docker/docker-common.sh` or in the IMAGE_VERSION env variable before running `dev-tools/src/main/docker/docker-deploy.sh`

**Comment by CloudResearch** [comment]
Hi @lorthirk ,

error is not fixed, after normal startup this error will be thrown:
```
Caused by: java.lang.IllegalArgumentException: Inet address may not be null
broker_1         |      at org.apache.http.util.Args.notNull(Args.java:54)
broker_1         |      at org.apache.http.HttpHost.<init>(HttpHost.java:170)
broker_1         |      at org.eclipse.kapua.service.datastore.client.rest.EsRestClientProvider.getClient(EsRestClientProvider.java:237)
broker_1         |      at org.eclipse.kapua.service.datastore.client.rest.EsRestClientProvider.<init>(EsRestClientProvider.java:217)
broker_1         |      ... 83 more

```
Its again the missing variable..

**Comment by lorthirk** [comment]
Is the Elasticsearch container running?

**Comment by CloudResearch** [comment]
Hi @lorthirk,

i'll restart my docker to be safe. Any thoughts about using some service like consul.io for service discovery (https://github.com/hashicorp/consul)?

**Comment by lorthirk** [comment]
Well... We could consider something like that for future developments, but it's certainly not needed for our reference deployment. Also, I don't think it's relevant to this issue... or I'm missing something perhaps?

**Comment by lorthirk** [comment]
So, I did a quick test today wiping out my entire local Docker images library and just using the `docker-deploy.sh` script, exporting the env variable IMAGE_VERSION=1.0.0-M2 before firing up the script. The images, with the correct version, downloaded successfully and are running without any error reported in the logs.

As I said before, Elasticsearch needs quite a bit of memory when running in a Docker container. In my Docker Settings I set 6 GBs per image, otherwise Elasticsearch would exit due to error 139 after a few seconds. May I suggest you to try the same?
